core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -213,13 +225,13 @@ private[spark] class CoarseMesosSchedulerBackend(
         val slaveId = offer.getSlaveId.toString
         val mem = getResource(offer.getResourcesList, "mem")
         val cpus = getResource(offer.getResourcesList, "cpus").toInt
+        val cpusToUse = math.min(cpus, maxCores - totalCoresAcquired)
```

Comment:
This doesn't handle your second case right? If `cpusToUse` is 5 and `spark.task.cpus` is 2, then there will be 1 core wasted on the executor. Do we need another check?